languageServer/src/features/StageCompletions.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  CompletionItem,
+  CompletionItemKind,
+  TextDocumentPositionParams
+} from 'vscode-languageserver/node'
+import { BaseFeature } from './BaseFeature'
+
+export class StageCompletions extends BaseFeature {
+  protected setup(): void {
+    this.connection?.onCompletion(
+      (params: TextDocumentPositionParams): CompletionItem[] => {
+        const stageNames = this.query({
+          dvcYamls: '$.workspace[?(@.uri.match(/dvc.yaml$/))]',
+          stages: '$.dvcYamls[*].parsed.stages.*~'
+        }).getValue('stages') as string[]
+
+        return stageNames.map(stage => {
+          return {
+            kind: CompletionItemKind.Method,
+            label: stage
```

Comment:
And then wrap them in a minimal completion item for this first case

Reply:
Did you build the `vsix` to demo this? I'm unable to get any completions when running the `dev-server` (Disabled the red-hat extension).

Reply:
No, I'll include it in today's tasks, but it shouldn't affect the server. I'll try it in the demo project today, I was testing with the getting-started project

Reply:
I was using example-get-started